src/reducers/fileTree.spec.tsx

```diff
@@ -38,20 +43,48 @@ describe(__filename, () => {
   describe('reducer', () => {
     it('builds and loads a tree', () => {
       const version = createInternalVersion(fakeVersion);
-      const state = reducer(undefined, actions.buildTree({ version }));
+      const comparedToVersionId = nextUniqueId();
+      const state = reducer(
+        undefined,
+        actions.buildTree({ comparedToVersionId, version }),
+      );
 
       expect(state).toEqual({
         ...initialState,
+        comparedToVersionId,
         forVersionId: version.id,
         tree: buildFileTree(version),
       });
     });
+
+    it('invalidates the tree when reloading a version', () => {
```

Comment:
I tried a bit and did not see infinite dispatch loop, but `buildTree` was dispatched one more time (because loading the base version results in an unnecessary reset of the file tree). So I guess it is better to keep that if statement. I added the test for reloading another version.